src/averageAges.js

```diff
@@ -54,9 +64,24 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
-}
+const calculateAverageAgeDiff = (people, onlyWithSon) => {
+  const ageDiff = onlyWithSon ? people.filter(({ sex, mother }) => sex === 'm'
+    && people.some(({ name }) => name === mother))
+    : people.filter(({ mother }) => people.some(({ name }) => name === mother));
+
+  ageDiff.map(child => {
+    const mother = people.find(({ name }) => name === child.mother);
+
+    child.motherAgeInBirth = child.born - mother.born;
+
+    return child;
+  });
```

Comment:
map returns a new array. you are using it here to modify an already existing array. you are also mutating the parameter (child)